pkg/ccl/changefeedccl/sink.go

```diff
@@ -581,3 +597,17 @@ func (s *safeSink) Flush(ctx context.Context) error {
 	defer s.Unlock()
 	return s.wrapped.Flush(ctx)
 }
+
+type SinkWithEncoder interface {
+	Sink
+
+	EncodeAndEmitRow(
+		ctx context.Context,
+		updatedRow cdcevent.Row,
+		topic TopicDescriptor,
+		updated, mvcc hlc.Timestamp,
+		alloc kvevent.Alloc,
+	) error
+
+	Flush(ctx context.Context) error
```

Comment:
I wonder if we need to introduce another sink type like this; What this says is SinkWithEncoder is a Sink (i.e. has EncodeRow methods, ) and also has flush method.  We are forcing other sinks to implement this method (e.g. error wrapper sink), and we're forcing parquet sink to implement regular sink methods (e.g. EncodeRow) as method returning an error.

What if instead of making this interface, you just made it something like:
```
type ParquetSink interface {
   EncodeAndEmitRow(
		ctx context.Context,
		updatedRow cdcevent.Row,
		topic TopicDescriptor,
		updated, mvcc hlc.Timestamp,
		alloc kvevent.Alloc,
	) error
```
You don't need to talk about other methods;  an implementation may optionally implement this method, and suddenly, it becomes parquet sink.
You already test (or at least should test) if the sink is a parquet sink, ie `if sink.(ParquetSink); ok {....}` -- so, that works fine.
On top of this, parquetSink can simply embed underlying `cloudStorageSink` -- so, suddenly, parquet sink is both a Sink and a ParquetSink.

Finally, if you do this, I think you'll be able to support resolved timestamp in parquet sink -- afterall, resolve files are just files written to cloud storage -- and you can do that w/out effecting parquet files.